subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -32,6 +32,10 @@
 #include <mgmt/mcumgr/transport/smp_internal.h>
 #endif
 
+#ifdef CONFIG_NRF_MCUBOOT_BOOT_REQUEST
```

Comment:
Rather than make no-up changes here, this file (with changes) should be moved to sdk-nrf, under the same path as it is here, then in CMake you can include that file in the build instead of this one when the Kconfig is set and you can prevent the file here being used by enabling `HEADER_FILE_ONLY` on it - see https://github.com/nrfconnect/sdk-nrf/blob/1cee1ccc3b1833dcbfad9aeb5166e2349b69e788/tests/subsys/net/lib/nrf_cloud/fota_common/CMakeLists.txt#L40 for an example, use `zephyr_library_amend` to append to the existing CMake library